File: php-laraval-app/app/Models/Store.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\BelongsToMany;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\SoftDeletes;
use Str;

class Store extends Model
{
    use HasFactory;
    use SoftDeletes;

    /**
     * Indicates if the model should be timestamped.
     *
     * @var bool
     */
    public $timestamps = true;

    /**
     * The attributes that are mass assignable.
     *
     * @var array
     */
    protected $fillable = [
        'address_id',
        'name',
        'company_number',
        'phone',
        'dial_code',
        'email',
    ];

    /**
     * The accessors to append to the model's array form.
     *
     * @var array
     */
    protected $appends = [
        'formatted_phone',
    ];

    /**
     * The attributes that should be cast.
     *
     * @var array
     */
    protected $casts = [
        'created_at' => 'datetime',
        'updated_at' => 'datetime',
        'deleted_at' => 'datetime',
    ];

    /**
     * Define the columns from current model that is needed by the accessor.
     *
     * @var array<string,string[]>
     */
    protected array $accessorsFields = [
        'formatted_phone' => ['dial_code', 'phone'],
    ];

    /**
     * Get the formatted customer's phone number.
     */
    public function getFormattedPhoneAttribute(): string
    {
        if (! $this->phone || $this->phone === $this->dial_code) {
            return '';
        }

        if (Str::startsWith($this->phone, $this->dial_code)) {
            return Str::replaceFirst($this->dial_code, "+{$this->dial_code} ", $this->phone);
        }

        return "+{$this->dial_code} {$this->phone}";
    }

    public function address(): BelongsTo
    {
        return $this->belongsTo(Address::class)->withTrashed();
    }

    public function products(): BelongsToMany
    {
        return $this->belongsToMany(Product::class, 'store_products')->withTimestamps();
    }

    public function users(): BelongsToMany
    {
        return $this->belongsToMany(User::class, 'store_users');
    }

    public function laundryOrders(): HasMany
    {
        return $this->hasMany(LaundryOrder::class);
    }

    public function sales(): HasMany
    {
        return $this->hasMany(StoreSale::class);
    }
}

File: dotnet/src/Downstairs.Infrastructure/Persistence/Models/Store.cs
namespace Downstairs.Infrastructure.Persistence.Models;

public partial class Store
{
    public long Id { get; set; }

    public long AddressId { get; set; }

    public string Name { get; set; } = null!;

    public string CompanyNumber { get; set; } = null!;

    public string? Phone { get; set; }

    public string? DialCode { get; set; }

    public string? Email { get; set; }

    public DateTime? CreatedAt { get; set; }

    public DateTime? UpdatedAt { get; set; }

    public DateTime? DeletedAt { get; set; }

    public virtual Address Address { get; set; } = null!;

    public virtual ICollection<CashierAttendance> CashierAttendances { get; set; } = new List<CashierAttendance>();

    public virtual ICollection<LaundryOrder> LaundryOrders { get; set; } = new List<LaundryOrder>();

    public virtual ICollection<StoreProduct> StoreProducts { get; set; } = new List<StoreProduct>();

    public virtual ICollection<StoreSale> StoreSales { get; set; } = new List<StoreSale>();

    public virtual ICollection<StoreUser> StoreUsers { get; set; } = new List<StoreUser>();

    public virtual ICollection<SubscriptionLaundryDetail> SubscriptionLaundryDetails { get; set; } = new List<SubscriptionLaundryDetail>();
}